apps/internal/local/server.go

```diff
@@ -145,7 +149,11 @@ func (s *Server) handler(w http.ResponseWriter, r *http.Request) {
 		desc := html.EscapeString(q.Get("error_description"))
 		// Note: It is a little weird we handle some errors by not going to the failPage. If they all should,
 		// change this to s.error() and make s.error() write the failPage instead of an error code.
-		_, _ = w.Write([]byte(fmt.Sprintf(failPage, headerErr, desc)))
+		if len(s.optionErrorPage) > 0 {
+			_, _ = w.Write(s.optionErrorPage)
+		} else {
+			_, _ = w.Write([]byte(fmt.Sprintf(failPage, headerErr, desc)))
```

Comment:
@chlowell 
AFAIK, there was no official specs on this due to the "custom error page" feature was not mandatory across all MSALs. But there was a CVE mentioning a potential XSS risk. In short, an implementation shall escape the error code and error description so that a malicious `error_code=<script>alert("this code snippet was executed")</script>` input shall not be executed.

@wayneforrest , I am not sure whether your "4: I have altered default fail-page also to be a html template (using .Code and .Err template variables)" can address that XSS risk.

Reply:
Hi @rayluo I did a quick test by updating the tests as follows:

`url.Values{"error": []string{"<script>alert('this code snippet was executed')</script>"}, "error_description": []string{"error_description"}},`

This resulted in the following fail page content: 

`"\n<!DOCTYPE html>\n<html>\n<head>\n    <meta charset=\"utf-8\" />\n    <title>Authentication Failed</title>\n</head>\n<body>\n\t<p>Authentication failed. You can return to the application. Feel free to close this browser tab.</p>\n\t<p>Error details: error &lt;script&gt;alert(&#39;this code snippet was executed&#39;)&lt;/script&gt;, error description: error_description</p>\n</body>\n</html>\n"`

The template variables got escaped. This page here also has examples: https://pkg.go.dev/html/template

I also made a small example here: https://go.dev/play/p/yfuM0RodQCh

Reply:
@wayneforrest , the escaping effect is what we want, so, if it is somehow already escaping properly, we don't have to fix anything. That being said, I would encourage you to find out where that escaping effect comes from, and then add comment to leave a hint to future maintainer that the escaping is happening. Currently, in this PR, I only saw you removed a pre-existing `html.EscapeString` and I did not dive deep to figure out why you still observed the escaping effect.

Regardless, can you also add a test case to make sure the escaping happens?

Reply:
@rayluo, I have added some comments in the code - most importantly is the import statement, as this can easily be altered incorrectly. 

I have also added a test, that should also guard against regression.